lib/puppet/provider/splunkforwarder_input/ini_setting.rb

```diff
@@ -7,9 +7,11 @@
   def self.file_path
     case Facter.value(:osfamily)
     when 'windows'
-      'C:\Program Files\SplunkUniversalForwarder\etc\system\local\inputs.conf'
+      FileUtils::mkdir_p 'C:\Program Files\SplunkUniversalForwarder\etc\apps\search\local'
```

Comment:
I really don't love mkdir -p being called every single time the provider is run. Why manage this directory here instead of in a manifest? Does the package installation not handle it?

Worst case scenario, it should only mkdir -p if the directory doesn't exist, but even that would be discouraged as it is a system change that would happen even on --noop puppet runs...